guppylang/std/option.py

```diff
@@ -1,81 +1,68 @@
-from collections.abc import Sequence
+from pathlib import Path
 from typing import Generic, no_type_check
 
-import hugr.tys as ht
-
-from guppylang.decorator import guppy
-from guppylang.error import InternalGuppyError
+from guppylang.decorator import ModuleIdentifier, guppy
+from guppylang.std import builtins
 from guppylang.std._internal.compiler.option import (
     OptionConstructor,
     OptionTestCompiler,
     OptionUnwrapCompiler,
     OptionUnwrapNothingCompiler,
 )
-from guppylang.std.builtins import mem_swap, owned
-from guppylang.tys.arg import Argument, TypeArg
-from guppylang.tys.param import TypeParam
-
-
-def _option_to_hugr(args: Sequence[Argument]) -> ht.Type:
-    match args:
-        case [TypeArg(ty)]:
-            return ht.Option(ty.to_hugr())
-        case _:
-            raise InternalGuppyError("Invalid type args for Option")
+from guppylang.std.builtins import L, mem_swap, owned
+from guppylang.tys.builtin import option_type_def
 
-
-T = guppy.type_var("T", copyable=False, droppable=False)
+builtins_module = guppy.get_module(
+    ModuleIdentifier(Path(builtins.__file__), "builtins", builtins)
+)
```

Comment:
This is hack to make `Option` a builtin type without moving it into the `builtins` module to cause less breakage.

We want to break up `builtins.py` anyways (see #732), so we can find a better solution once we tackle #732/